### PR TITLE
[3.11] gh-102304: doc: Add links to Stable ABI and Limited C API (#105345) (#105371)

### PR DESCRIPTION
**Pull request by AA-Turner**
* gh-102304: doc: Add links to Stable ABI and Limited C API (#105345)

* Add "limited-c-api" and "stable-api" references.
* Rename "stable-abi-list" reference to "limited-api-list".
* Makefile: Document files regenerated by "make regen-limited-abi"
* Remove first empty line in generated files:

  - Lib/test/test_stable_abi_ctypes.py
  - PC/python3dll.c

(cherry picked from commit bae415ad02c79cf3a2eec4aa6969221a12e6716f)

* gh-102304: Fix up Simple ABI doc (GH-105351)

(cherry picked from commit 0202aa002e06acef9aa55ace0d939103df19cadd) (cherry picked from commit 82ab13c49a77b0d73e6cd5c7ecfe0de1b1036819)

-----

These targets were added for 3.13 and 3.12, but they're needed for the devguide: https://github.com/python/devguide/pull/1167

A

----
:books: Documentation preview :books:: https://cpython-previews--109901.org.readthedocs.build/

### CODE REVIEW

**Comment by erlend-aasland** [comment]
Unfortunately, bae415ad02c79cf3a2eec4aa6969221a12e6716f does a lot of things:

- doc updates
- C source updates
- script updates

AFAICS, only the doc updates are relevant for python/devguide#1167? @vstinner, should the whole commit be backported, or only parts of it?

**Comment by vstinner** [comment]
It's up to you, I don't know. I understood that the doc should look the same in all branches: all doc changes should be backported.

**Comment by AA-Turner** [comment]
> should the whole commit be backported, or only parts of it?

I'd intended to only backport the documentation changes; something went wrong in preparing this PR. But if we want to backport everything happy to update the PR, I just wanted to err on the cautious side.

A

**Comment by erlend-aasland** [comment]
> It's up to you, I don't know. I understood that the doc should look the same in all branches: all doc changes should be backported.

The problem is that the **original commit** is not a **doc change**; it changes a lot of non-doc stuff.

File: Doc/c-api/stable.rst [hunk 1]
```diff
@@ -147,12 +160,12 @@ This is the case with Windows and macOS releases from ``python.org`` and many
 third-party distributors.
 
 
-.. _stable-abi-list:
+.. _limited-api-list:
```
Comment (erlend-aasland) [inline Doc/c-api/stable.rst:163]: This looks weird; there's already such an anchor in line 171.

Comment (erlend-aasland) [reply Doc/c-api/stable.rst:163]: Ah, a _similar_ anchor in 171; not equal.

Comment (erlend-aasland) [reply Doc/c-api/stable.rst:163]: So, what is the `_limited-api-list` anchor used for then?

Comment (AA-Turner) [reply Doc/c-api/stable.rst:163]: There are 4 cross-references (one in ``stable.rst`` (L40), one in 3.11 whatsnew, and two in the 3.11 changelog) -- search for ``[`<]limited-api-list[`>]``.

A

Comment (AA-Turner) [reply Doc/c-api/stable.rst:163]: > Ah, a *similar* anchor in 171; not equal.

Oh, I think I understand. L171 is a custom directive also named ``limited-api-list``, to enumerate the limited API -- see [the rendered page](https://docs.python.org/dev/c-api/stable.html#contents-of-limited-api) for an example. So both are valid.

Comment (erlend-aasland) [reply Doc/c-api/stable.rst:163]: Ok, thanks!

**Review by erlend-aasland — APPROVED** [review]

**Comment by erlend-aasland** [comment]
@AA-Turner, could you write comment that sums up this change (for the merge commit message)?

**Comment by AA-Turner** [comment]
First draft:

----

Add links to the Stable ABI and Limited C API

* Add "limited-c-api" and "stable-api" targets
* Rename the "stable-abi-list" target to "limited-api-list"

----

A

**Review by vstinner — APPROVED** [review]
LGTM

**Comment by erlend-aasland** [comment]
> First draft:

Looks ok to me!

**Pull request merged**